### PR TITLE
Add a sqlite DB to the browser

### PR DESCRIPTION
**Pull request by habdelra**
This runs a test that proves a sqlite DB is running in the browser of the host app (in tests).

### CODE REVIEW

File: packages/host/tests/unit/sqlite-test.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    // https://webkit.org/blog/12257/the-file-system-access-api-with-origin-private-file-system/
+
+    const db = new sqlite.oo1.DB({ filename: ':memory:' });
+
+    try {
+      db.exec(`
+        CREATE TABLE t(a,b);
+        INSERT INTO t(a,b) VALUES('abc',123),('def',456),(NULL,789),('ghi',012);
+      `);
+      let results = query(db, `SELECT * FROM t;`);
```
Comment (habdelra) [inline packages/host/tests/unit/sqlite-test.ts]: its quite fascinating that the sqlite API for executing SQL is synchronous. the downside is that it's blocking in the main thread while it executes SQL. Perhaps we should also explore how to run this in a worker (not service worker) such that we can query the DB in the main thread, have it execute in a worker, and return the result back to the caller in the main thread.

Comment (habdelra) [reply packages/host/tests/unit/sqlite-test.ts]: I'll explore a worker implementation in the next commit i post to this PR...

**Review by lukemelia — APPROVED** [review]
Remarkably simple integration!

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -25039,3 +25043,7 @@ packages:
     dependencies:
       eslint: 8.37.0
     dev: true
+
+settings:
+  autoInstallPeers: true
+  excludeLinksFromLockfile: false
```
Comment (backspace) [inline pnpm-lock.yaml:25049]: What’s your pnpm version? I’m confused about why this keeps happening.

**Review by backspace — APPROVED** [review]
wow!

**Review by tintinthong — APPROVED** [review]

File: pnpm-lock.yaml (hunk 2 above)
Comment (habdelra) [reply pnpm-lock.yaml:25049]: My pnpm version is 8.5.1. i just noticed that we have a volta assertion for pnpm 8.10.5.

Comment (habdelra) [reply pnpm-lock.yaml:25049]: I updated to pnpm 8.15.4 and re-ran `pnpm install` and it made no changes to the lock file.

**Pull request merged**